string_extra.c

```diff
@@ -0,0 +1,33 @@
+#include <stdlib.h>
+#include <string.h>
+
+// Public domain strtok_r() by Charlie Gordon
+char* strtok_r(char *str, const char *delim, char **nextp)
```

Comment:
I'm sure I've mentiond this somewhere before, but you can also discuss this upstream at https://github.com/DevSolar/pdclib , so that the pdclib internal functions can possibly expose a re-usable interface.

*(As this would delay this PR for longer, I suggest to still accept this change - then go upstream and try to find out if `__private__strtok_r` might be an upstream-option, to reduce the implementation here)*

Reply:
This is surely a good idea, when I'll have some free time I'll open a ticket in the pdclib repo asking about the possibility to expose strtok_r function.